examples/run_CPMG.py

```diff
@@ -4,149 +4,184 @@
 
 This script will perform a standard CPMG experiment. 
 In order to form a symmetric echo, a padding time is added before 
-and after your tau through a series of delays. 
+and after your tau through a series of delays.
+If you wish to keep the field as is without adjustment, follow
+the 'py run_CPMG.py' command with 'stayput' (e.g. 'py run_CPMG.py stayput')
 """
 from pylab import *
 from pyspecdata import *
+import os, sys
 from numpy import *
 import SpinCore_pp
-from SpinCore_pp.ppg import run_cpmg
-import os
+from SpinCore_pp import prog_plen, get_integer_sampling_intervals
+from SpinCore_pp.ppg import generic
 from datetime import datetime
+from Instruments.XEPR_eth import xepr
 import h5py
-raise RuntimeError("This pulse proram has not been updated.  Before running again, it should be possible to replace a lot of the code below with a call to the function provided by the 'generic' pulse program inside the ppg directory!")
 
-fl = figlist_var()
 # {{{importing acquisition parameters
 config_dict = SpinCore_pp.configuration("active.ini")
-nPoints = int(config_dict["acq_time_ms"] * config_dict["SW_kHz"] + 0.5)
+(
+    nPoints,
+    config_dict["SW_kHz"],
+    config_dict["echo_acq_ms"],
+) = get_integer_sampling_intervals(
+    SW_kHz=config_dict["SW_kHz"], acq_time_ms=config_dict["echo_acq_ms"]
+)
+my_exp_type = "ODNP_NMR_comp/CPMG"
+target_directory = getDATADIR(exp_type=my_exp_type)
+assert os.path.exists(target_directory)
+# }}}
+# {{{ command-line option to leave the field untouched (if you set it once, why set it again)
+adjust_field = True
+if len(sys.argv) == 2 and sys.argv[1] == "stayput":
+    adjust_field = False
+# }}}
+# {{{ let computer set field
+if adjust_field:
+    print(
+        "I'm assuming that you've tuned your probe to",
+        config_dict["carrierFreq_MHz"],
+        "since that's what's in your .ini file",
+    )
+    field_G = config_dict["carrierFreq_MHz"] / config_dict["gamma_eff_MHz_G"]
+    print(
+        "Based on that, and the gamma_eff_MHz_G you have in your .ini file, I'm setting the field to %f"
+        % field_G
+    )
+    with xepr() as x:
+        assert field_G < 3700, "are you crazy??? field is too high!"
+        assert field_G > 3300, "are you crazy?? field is too low!"
+        field_G = x.set_field(field_G)
+        print("field set to ", field_G)
```

Comment:
implement inline command to keep the field or adjust

Reply:
what does this comment mean?

Reply:
commenting on rational for new lines - this section takes the inline command so that the user has the option to "stayput" otherwise it sets the field